### PR TITLE
feat: create week2 study contents

### PR DESCRIPTION
**Pull request by hyeonjoo**

### CODE REVIEW

File: week2/NextJs.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+**이미지 리사이즈, 최적화, `WebP`와 같은 모던 포맷을 제공**합니다. 이것이 이미지 자동 최적화입니다.
+
+- WebP 는 PNG, JPEG 보다 컬러의 깊이감, 애니메이션 프레임, 불투명도 등에서 더 나은 압축 방식입니다. 크롬, 엣지, 파이어폭스, 오페라, 사파리에서 지원해요. ~~인터넷익스플로러 안녕~~
+- 화면의 표시 영역(viewport)이 작다면 그에 맞춰 굳이 큰 이미지를 보이지 않도록 합니다.
+- 외부에서 이미지를 가져오더라도 다 동작 합니다.
+
+`Webpack` 의 경우 이미지 최적화는 빌드 시에 수행되는데, `Next.js` 에서는 유저가 **요청**할 때, 즉 수요에 맞추어 최적화 합니다. 따라서 일반적인 정적 사이트를 만드는 툴과 다르게 `Next.js` 의 빌드 시간은 이미지 개수에 영향을 받지 않습니다.
+
+이미지는 디폴트로 **lazy loading** 됩니다. 스크롤해서 viewport에 들어와 있는 이미지들이 로드됩니다.
```
Comment (56aiden90) [inline week2/NextJs.md:23]: https://css-tricks.com/the-complete-guide-to-lazy-loading-images/
https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API

I have no idea about how Nextjs implements image lazy loading internally but the links above might be interesting. Thanks!

Comment (56aiden90) [inline week2/NextJs.md:23]: https://css-tricks.com/the-complete-guide-to-lazy-loading-images/
https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API

I have no idea about how Nextjs implements image lazy loading internally but the links above might be interesting. Thanks!

**Review by JeGwan — COMMENTED** [review]
다음에 CSR인경우 Next.js가 초기 HTML 바디를 어떻게 주는지도 다뤄주시면 좋을 것 같아요!

File: week2/NextJs.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+- **첫 요청시에 한 페이지만** 불러오게 됩니다.
+- **초기 로딩 이후로 웹사이트 렌더링이 빠릅니다.**
+- 사이트 간 상호 작용도 활발하고, 자바스크립트 라이브러리를 잘 활용하기 때문에 유저 모션에도 민감하게 반응할 수 있죠.
+- 하지만 단점이 있습니다.
+  - **초기 페이지 로딩이 느립니다.**
+    - html이 있고, 그 위에 javascript 코드가 올라가고, 이 javascript 코드가 실행되고, 그 코드 안에서 서버 요청도 대기타야되고, UI 꾸미는 코드 있으면 그것도 기다려야 하기 때문입니다.
+    - 대부분 추가적인 라이브러리를 필요로 합니다.
+  - **검색 엔진 최적화가 어렵습니다.**
+    - 검색 엔진이 response로 받게 되는 사이트들이 js로 렌더링 되어야 하는 상태, 컨텐츠가 텅 빈상태니까, 이부분에서 최적화가 어렵습니다.
```
Comment (jordan-choi) [inline week2/NextJs.md:168]: 관련 글 아래에 남깁니다. 
The page that you have created with the latest technologies and utmost care might seem like a blank page to Google [1]. 
[1] https://zeo.org/blog/javascript-seo-what-is-ssr-csr-advantages-and-disadvantages/

**Pull request merged**